### PR TITLE
Better unknown namer errors

### LINKED ISSUE #58: hard-to-understand error with non-existing namers
If you put a namer in the dtab that linkerd doesn't know about, you get the following behavior:

* The log says something unintelligible:
```
E 0201 20:23:21.968 THREAD19: Exception propagated to the root monitor!
Failure(io.l5d.http.anyMethodPfx, flags=0x100000000)
	with Service -> 0.0.0.0/4140
	at com.twitter.finagle.NoStacktrace(Unknown Source)
```
* The name of the unknown namer is returned as a text string with a 503 error (at least, with HTTP--haven't tried other protocols).

Ideally, the log message would be more informative. For the HTTP error, the 503 part is good, but ideally, the content would be either nothing or a more descriptive error message.

### PR DESCRIPTION
**Pull request by esbie**
The global namer's lookup method thinks anything in the pattern of `/$/foo` is a class ... meaning a mistyped namer can result in a ClassNotFoundException.

Here I'm wrapping the exception to provide additional context

Fixes #58

### CODE REVIEW

File: linkerd/core/src/main/scala/io/buoyant/linkerd/DefaultInterpreterInitializer.scala [hunk 1]
```diff
@@ -48,7 +50,14 @@ case class ConfiguredNamersInterpreter(namers: Seq[(Path, Namer)])
       // For now, punt to the default interpreter. Later, we
       // should consider removing the finagle Namer.global, which
       // provides /$/ names.
-      case Nil => Namer.global.lookup(path)
+      case Nil =>
+        Namer.global.lookup(path).transform {
+          case Activity.Failed(e) if e.isInstanceOf[ClassNotFoundException] =>
+            Activity.exception(UnknownNamer(path, e))
+          case Activity.Failed(e) => Activity.exception(e)
```
Comment (adleong) [inline linkerd/core/src/main/scala/io/buoyant/linkerd/DefaultInterpreterInitializer.scala]: could all of these be grouped into a passthrough like 
```
case act => act
```
?

File: linkerd/core/src/main/scala/io/buoyant/linkerd/DefaultInterpreterInitializer.scala [hunk 2]
```diff
@@ -48,7 +50,14 @@ case class ConfiguredNamersInterpreter(namers: Seq[(Path, Namer)])
       // For now, punt to the default interpreter. Later, we
       // should consider removing the finagle Namer.global, which
       // provides /$/ names.
-      case Nil => Namer.global.lookup(path)
+      case Nil =>
+        Namer.global.lookup(path).transform {
```
Comment (adleong) [inline linkerd/core/src/main/scala/io/buoyant/linkerd/DefaultInterpreterInitializer.scala]: It is super lame that `Activity` doesn't have a `rescue` method.

File: linkerd/core/src/main/scala/io/buoyant/linkerd/DefaultInterpreterInitializer.scala (hunk 1 above)
Comment (esbie) [inline linkerd/core/src/main/scala/io/buoyant/linkerd/DefaultInterpreterInitializer.scala]: I'm not an activity expert, but transform expects matches on `Activity.State` that map to `Activity` :|

File: linkerd/core/src/main/scala/io/buoyant/linkerd/DefaultInterpreterInitializer.scala (hunk 2 above)
Comment (esbie) [inline linkerd/core/src/main/scala/io/buoyant/linkerd/DefaultInterpreterInitializer.scala]: yah, I wanted to use `.handle{}`, but it expects you to like... actually handle the error, not turn it into another better error

File: linkerd/core/src/main/scala/io/buoyant/linkerd/DefaultInterpreterInitializer.scala (hunk 1 above)
Comment (olix0r) [inline linkerd/core/src/main/scala/io/buoyant/linkerd/DefaultInterpreterInitializer.scala]: Can do:
```scala
case state => Activity(state)
```

File: linkerd/core/src/main/scala/io/buoyant/linkerd/DefaultInterpreterInitializer.scala [hunk 3]
```diff
@@ -48,7 +50,14 @@ case class ConfiguredNamersInterpreter(namers: Seq[(Path, Namer)])
...
+          case Activity.Failed(e) if e.isInstanceOf[ClassNotFoundException] =>
```
Comment (olix0r) [inline linkerd/core/src/main/scala/io/buoyant/linkerd/DefaultInterpreterInitializer.scala]: prefer:
```scala
case Activity.Failed(e: ClassNotFoundException) =>
```

**Comment by olix0r** [comment]
looks good! just some small style nits.

File: linkerd/core/src/main/scala/io/buoyant/linkerd/DefaultInterpreterInitializer.scala (hunk 3 above)
Comment (esbie) [inline linkerd/core/src/main/scala/io/buoyant/linkerd/DefaultInterpreterInitializer.scala]: cool, turns out I was unnecessarily worried about type erasure

**Comment by siggy** [comment]
:star: :+1: coool
```
$ curl http://localhost:4140
Unknown namer: FOOio.buoyant.http.anyMethodPfx for path: /$/FOOio.buoyant.http.anyMethodPfx/host/GET/localhost:4140
```

**Comment by adleong** [comment]
:star: :mahjong:

**Pull request merged**